content/technotes/2020-03-19-parzer/index.Rmd

```diff
@@ -0,0 +1,157 @@
+(https://twitter.com/noamross/status/1070733367522590721) about 15 months ago.
+
+The idea being that sometimes you have geographic coordinates in a messy
+format, or in many different formats, etc. You can think of it as being
+the package for geographic coordinates that [lubridate][] is for dates.
+
+I started off thinking about wrapping a Javascript library with [Jeroen's][jeroen]
+[V8][] R package, but then someone showed me or I found (can't remember) [some C++ code][cll]
+from back in 2006 that seemed appropriate. I figured I'd go down the C++
+track instead of the Javascript track because I figured I could likely get
+better performance out of C++ and have slightly less install headaches for users.
+
+
+Package installation
+------------------
+
+The package is on CRAN so you can use `install.packages`
+
+```{r eval=FALSE}
+install.packages("parzer")
+```
+
+However, since this package requires compilation you probably want a binary.
+Binaries are not available on CRAN yet. You can install a binary like
+
+```{r eval=FALSE}
+install.packages("parzer", repos = "https://dev.ropensci.org/")
+```
+
+```{r}
+library(parzer)
+```
+
+Check out the package documentation to get started: <https://docs.ropensci.org/parzer/>
+
+<br><br>
+
+Package basics
+------------------
+
+The following is a summary of the functions in the package and what they do:
+
+Parse latitude or longitude separately
+
+- parse_lat
+- parse_lon
+
+Parse latitudes and longitudes at the same time
+
+- parse_lon_lat
+
+Parse into separate parts of degrees, minutes, seconds
+
+- parse_parts_lat
+- parse_parts_lon
+
+Pull out separately degrees, minutes, seconds, or hemisphere
+
+- pz_degree
+- pz_minute
+- pz_second
+- parse_hemisphere
+
+Add/subtract degrees, minutes, seconds
+
+- pz_d
+- pz_m
+- pz_s
+
+Some examples:
+
+parse latitudes and longitudes
+
+```{r warning=TRUE}
+lats <- c("40.123°", "40.123N74.123W", "191.89", 12, "N45 04.25764")
+parse_lat(lats)
+longs <- c("45W54.2356", "181", 45, 45.234234, "-45.98739874N")
+parse_lon(longs)
+```
+
+In the above examples you can see there's a mix of valid coordinate
+values as well as invalid values. There's a mix of types supported as well.
+
+Sometimes you may want to parse a geographic coordinate into its component
+parts; `parse_parts_lat` and `parse_parts_lon` are what you need:
+
+```{r}
+x <- c("191.89", 12, "N45 04.25764")
+parse_parts_lon(x)
+```
+
+Taking a cue from lubridate, we thought it would be useful to make it easier
+to add or subtract numbers for coordinates. Three functions help with this:
+
+```{r collapse=TRUE}
+pz_d(31)
+pz_d(31) + pz_m(44)
+pz_d(31) - pz_m(44)
+pz_d(31) + pz_m(44) + pz_s(59)
+pz_d(-121) + pz_m(1) + pz_s(33)
+```
+
+<br><br>
+
+Use cases
+------------------
+
+Check out the [parzer use cases][usecases] vignette on the docs site.
+Get in touch if you have a use case that might be good to add to that
```

Comment:
how do you want people to get in touch?
add to https://discuss.ropensci.org/c/usecases/10 and tag with `parzer`? (and you can incorp into vignette?)

Reply:
via issues, will include a link